### PR TITLE
Add a new exoscale-sru template

### PR DESCRIPTION
**Pull request by chrisglass**
This script outlines what steps should be taken to kick the tires on
exoscale instances for cloud-init SRUs.

### CODE REVIEW

File: sru-templates/manual/exoscale-sru [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    case $series in
+        xenial) token="Ubuntu 16.04";;
+        bionic) token="Ubuntu 18.04";;
+        cosmic) token="Ubuntu 18.10";;
+        disco) token="Ubuntu 19.04";;
+        *) echo "Unknown series, update exoscale script!"; exit 1;;
+    esac
+
+    # find the template ID
+    template_id=$($exo vm template list --output-format text | grep $token | cut -f1)
```
Comment (raharper) [inline sru-templates/manual/exoscale-sru]: your token has spaces, you'll want to quote that:

```
(neipa) ~ % token="Ubuntu 19.04" 
(neipa) ~ % echo "FOo bar baz Ubuntu 19.04 wark" | grep "$token" 
FOo bar baz Ubuntu 19.04 wark
(neipa) ~ % echo "FOo bar baz Ubuntu 19.04 wark" | grep $token 
grep: 19.04: No such file or directory
     ..:(╯°□°)╯彡┻━━┻

```

File: sru-templates/manual/exoscale-sru [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- cloud-init analyze show
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- cloud-init analyze blame
+
+    # Upgrade to -proposed cloud-init and reboot
+    scp "${sshopts[@]}" setup_proposed.sh ubuntu@$VM_IP:.
+    ssh "${sshopts[@]}" ubuntu@$VM_IP sudo bash ./setup_proposed.sh 2>&1 | egrep 'cloud-init'
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- dpkg-query --show cloud-init
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- sudo hostname SRU-didnt-work
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- sudo cloud-init clean --logs --reboot
+    ssh-keygen -f "/home/ubuntu/.ssh/known_hosts" -R $VM_IP
```
Comment (raharper) [inline sru-templates/manual/exoscale-sru]: This isn't needed if you're including the sshopts which specify known-hosts file as /dev/null.

File: sru-templates/manual/exoscale-sru [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+    echo "Rebooted with the SRU version of cloud-init"
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- hostname
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- "grep Trace /var/log/cloud-init*"
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- systemd-analyze
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- systemd-analyze blame
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- cloud-init analyze show
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- cloud-init analyze blame
+
+    # Check that the private network interface is up - this should show *two* interfaces (3 with lo)
+    ssh "${sshopts[@]}" ubuntu@$VM_IP -- ifconfig -s
```
Comment (raharper) [inline sru-templates/manual/exoscale-sru]: This ifconfig isn't always present; I suggest using ``ip`` output, or /sys/class/net/* greping.

**Comment by chrisglass** [comment]
Hi, thanks for the time reviewing this branch so far. 

As you've discovered it's not (at all) ready for prime time yet - hence the "draft" status. 

Sorry for the inconvenience caused. It is probably better than I close this and repoen it as a real PR once it's finished baking.

**Comment by raharper** [comment]
> Hi, thanks for the time reviewing this branch so far.
> 
> As you've discovered it's not (at all) ready for prime time yet - hence the "draft" status.
> 
> Sorry for the inconvenience caused. It is probably better than I close this and repoen it as a real PR once it's finished baking.

None caused.  I was here putting up a new PR and saw yours, figured I'd leave some comments for whenever you pushed up something new.

**Comment by chrisglass** [comment]
Well, in that case I'll repoen it and schedule to push my changes sooner rather than later :)

**Pull request closed without merging**